packages/cactus-plugin-ledger-connector-polkadot/src/main/typescript/web-services/deploy-contract-ink-endpoint.ts

```diff
@@ -0,0 +1,107 @@
+import { Express, Request, Response } from "express";
+import HttpStatus from "http-status-codes";
+
+import {
+  Logger,
+  LoggerProvider,
+  LogLevelDesc,
+  Checks,
+  IAsyncProvider,
+} from "@hyperledger/cactus-common";
+
+import {
+  IWebServiceEndpoint,
+  IExpressRequestHandler,
+  IEndpointAuthzOptions,
+} from "@hyperledger/cactus-core-api";
+
+import { registerWebServiceEndpoint } from "@hyperledger/cactus-core";
+
+import { PluginLedgerConnectorPolkadot } from "../plugin-ledger-connector-polkadot";
+import { DeployContractInkEndpoint as Constants } from "./deploy-contract-ink-endpoint-constants";
+
+export interface IDeployContractInkOptions {
+  logLevel?: LogLevelDesc;
+  connector: PluginLedgerConnectorPolkadot;
+}
+
+export class DeployContractInkEndpoint implements IWebServiceEndpoint {
+  private readonly log: Logger;
+  public static readonly CLASS_NAME = "InvokeContractEndpoint";
+
+  constructor(public readonly opts: IDeployContractInkOptions) {
+    const fnTag = "RunTransactionEndpointV1#constructor()";
+
+    Checks.truthy(opts, `${fnTag} options`);
+    Checks.truthy(opts.connector, `${fnTag} options.connector`);
+
+    this.log = LoggerProvider.getOrCreate({
+      label: this.className,
+      level: opts.logLevel || "INFO",
+    });
+  }
+
+  public get className(): string {
+    return DeployContractInkEndpoint.CLASS_NAME;
+  }
+
+  getAuthorizationOptionsProvider(): IAsyncProvider<IEndpointAuthzOptions> {
+    // TODO: make this an injectable dependency in the constructor
+    return {
+      get: async () => ({
+        isProtected: true,
+        requiredRoles: [],
+      }),
+    };
+  }
+
+  public getExpressRequestHandler(): IExpressRequestHandler {
+    return this.handleRequest.bind(this);
+  }
+
+  public getPath(): string {
+    return Constants.HTTP_PATH;
+  }
+
+  public getVerbLowerCase(): string {
+    return Constants.HTTP_VERB_LOWER_CASE;
+  }
+
+  public getOperationId(): string {
+    throw new Error("Method not implemented.");
+  }
+
+  public get oasPath(): string {
+    throw new Error("Method not implemented.");
+  }
+
+  public async registerExpress(
+    expressApp: Express,
+  ): Promise<IWebServiceEndpoint> {
+    await registerWebServiceEndpoint(expressApp, this);
+    return this;
+  }
+
+  async handleRequest(req: Request, res: Response): Promise<void> {
+    const fnTag = "DeployContractInkEndpoint#handleRequest()";
+    this.log.debug(`POST ${this.getPath()}`);
+    try {
+      const message =
+        `${this.opts.connector.className} does not support ` +
```

Comment:
How are we tracking these TODO code?
(TODOs in the code help to track them easily, followed by some GH issues in place to fill these up later)

Reply:
I believe we can open up some issues referring to those as soon as the PR is merged